sky/packages/sky/lib/sky_tool

```diff
@@ -254,6 +284,355 @@ class StopSky(object):
         pids.clear()
 
 
+class IOSDevice(object):
+    _has_ios_deploy = None
+    @classmethod
```

Comment:
I think these are all working with caching state that is essentially just a set of globals tied to physical hardware in some way, so I felt like it would be cleaner to say IOSDevice.get_foo() throughout the code, rather than risking users accidentally newing up multiple IOSDevices and consequently going down the slow path.  That said, if class methods are bad in python, I'm not deeply wedded to this approach.